zkllvm/best-practices-limitations/rust-derive.mdx

```diff
@@ -0,0 +1,81 @@
+# Structs and enums in Rust
+
+Whenever a Rust circuit is compiled, `rustc` applies various optimizations to reduce its memory usage.
+
+Among these memory optimizations is [**reordering fields in structs and enums to avoid unnecessary 'paddings' in circuit IRs**](https://users.rust-lang.org/t/disable-struct-reordering-optimization-when-emitting-llvm-bitcode/74951). Consider the following example:
+
+```rust
```

Comment:
It's not necessary, but maybe for demonstration purposes we can construct smaller example? This one is a bit too big than needed to illustrate the issue and probably it will be a bit harder for users to get the point.

Reply:
Sure, I've simplified the example